### PR TITLE
feat: add edit function for companies page

### PR DESCRIPTION
**Pull request by yusualhashash**
## How has this been tested?

- [ ] Prod
- [ ] Development

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] Added all translations, no hard coded text. 
- [ ] Updated the UI (if needed).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by eren-ayasofyazilim — APPROVED** [review]

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      .catch((error: ApiError) => {
+        return {
+          data: error,
+          status: error.status,
+        };
+      }),
+  );
+  revalidatePath("/");
+  return response;
+}
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts:34]: We should turn this into async/await and use structured error which is written by @ertugrulcan-ays and add the return type to the function.

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  } else if (data === "taxOffices") {
+    response = await client.taxOffice.getApiCrmServiceTaxOfficesDetailById({
+      id: body.id,
+    });
+  }
+  revalidatePath("/");
+  return {
+    data: response,
+    status: 200,
+  };
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts:60]: what if we had an error from the backend? wrapping this with try catch and using return type would be great.

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        id,
+        organizationId,
+        requestBody,
+      },
+    );
+    revalidatePath("/");
+    return true;
+  } catch (error) {
+    return error;
+  }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts:82]: using structured error would be great here.

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  return {
+    data: response,
+    status: 200,
+  };
+}
+export async function updateMerchantCRMDetailServer(
+  id: string,
+  organizationId: string,
+  type: "merchant",
+  requestBody: UniRefund_CRMService_Organizations_UpdateOrganizationDto,
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts:66]: we agreed on using object params for the functions, let's turn this into that style and any future style that we are going ot have.

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+      await client.telephoneType.putApiCrmServiceTelephonesById({
+        id: requestBody.id,
+        requestBody,
+      });
+    }
+    revalidatePath("/");
+    return true;
+  } catch (error) {
+    return error;
+  }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts:113]: Using structred error would be awesome.

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  try {
+    const client = await getCRMServiceClient();
+    await client.telephoneType.putApiCrmServiceTelephonesById({
+      id: requestBody.id,
+      requestBody,
+    });
+    revalidatePath("/");
+    return true;
+  } catch (error) {
+    return error;
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts:128]: using structrued error would be awesome

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    const client = await getCRMServiceClient();
+    await client.addressType.putApiCrmServiceAddressesById({
+      id: requestBody.id,
+      requestBody,
+    });
+    revalidatePath("/");
+    return true;
+  } catch (error) {
+    return error;
+  }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/action.ts:145]: Structured error

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/form.tsx [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+      await updateCRMDetailServer({
+        ...values,
+        id: addressInfo?.id,
+      } as UniRefund_CRMService_AddressTypes_UpdateAddressTypeDto);
+      response = "success";
+    }
+
+    if (response) {
+      toast.success("Updated successfully!");
+    }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/form.tsx:116]: We always return the response as successful even if it was fail!

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/form.tsx [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+      } as UniRefund_CRMService_TelephoneTypes_UpdateTelephoneTypeDto);
+      response = "success";
+    }
+    if (sectionName === "address") {
+      await updateCRMDetailServer({
+        ...values,
+        id: addressInfo?.id,
+      } as UniRefund_CRMService_AddressTypes_UpdateAddressTypeDto);
+      response = "success";
+    }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/form.tsx:112]: Did not we have such structure in the server side>?

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(main)/page.tsx [hunk 10]
```diff
@@ -275,27 +188,17 @@ export default function Page({
...
-    autoFormArgs: autoformEditArgs,
-    callback: (data, row) => {
-      onEdit(data, row, editFormSchema);
-    },
-  });
+
   const router = useRouter();
   columnsData.data.actionList?.push({
-    cta: " details",
+    cta: "Edit",
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(main)/page.tsx:194]: We can use the language model here1

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/page.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  if (data === "merchants") {
+    try {
+      const response = (await getCRMMerchantDetailServer({ id })).data;
+      if ("merchant" in response) {
+        return response.merchant;
+      }
+      return null;
+    } catch (error) {
+      return null;
+    }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/page.tsx:15]: Let's get the error here.

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/page.tsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  try {
+    const response = (await getCRMDetailServer(data, { id })).data;
+    return response;
+  } catch (error) {
+    return null;
+  }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/page.tsx:23]: let's catch the error and return it.

File: apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/page.tsx [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+  params: {
+    id: string;
+    data: string;
+    domain: string;
+  };
+}) {
+  const crmDetailData = await getCrmDetailData(params.data, params.id);
+  if (!crmDetailData) {
+    return <>Not found</>;
+  }
```
Comment (a0m0rajab) [inline apps/web/src/app/[lang]/app/[type]/crm/[domain]/[data]/(alternative)/details/[id]/page.tsx:37]: a better style could be provided.

**Review by a0m0rajab — COMMENTED** [review]
Minor changes mostly around: 
- [ ] using structured error 
- [ ] turning the error to the user on the UI 
- [ ] improving the code

**Pull request merged**